### ISSUE TITLE
GD-351: `C#` Not all tests can be executed, the run stops at some point

### ISSUE THREAD

**Issue by Delsin-Yu** [MikeSchulze/gdUnit4#351]
### The used GdUnit4 version

4.2.0 (Latest Release)

### The used Godot version

Godot v4.2.stable.mono [Official Build]

### Operating System

Godot v4.2.stable.mono - Windows 10.0.17763 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 4060 Ti (NVIDIA; 31.0.15.4601) - AMD Ryzen 9 5900X 12-Core Processor (24 Threads)

### Describe the bug

|GDUnit4 freezes in the middle of running tests for my project.|Running the test in debug mode shows the following error
|-|-
|![image](https://github.com/MikeSchulze/gdUnit4/assets/71481700/c6f24a9a-2bf0-47c5-98ea-8c0768e9172e)|![image](https://github.com/MikeSchulze/gdUnit4/assets/71481700/4a33abd1-fb52-461c-9d6e-dab3f73673cf)

### Steps to Reproduce

0. Environment Setup: `.Net 8 SDK` is required.
1. Clone the source code for the [`1.0.0` release of `GDTask.Nuget`](https://github.com/Delsin-Yu/GDTask.Nuget/releases/tag/v1.0.0) (commit [4412a7d](https://github.com/Delsin-Yu/GDTask.Nuget/commit/4412a7d63787e0e259ba258d5d8cdaf3e001d608))
2. Open the `./GDTask.Tests` directory with the official build of Godot Editor `v4.2.0 stable.mono`
3. From the FileSystem, right-click the `res://test/` directory, and select `Run Testsuites`.
4. Expecting the running tests stops at `GDTaskTest_Observable`.

### Minimal reproduction project

Fork or Clone the [GDTask.Nuget](https://github.com/Delsin-Yu/GDTask.Nuget) repo and switch to the commit labeled with `v1.0.0`, or [download the source code for the Release v1.0.0](https://github.com/Delsin-Yu/GDTask.Nuget/archive/refs/tags/v1.0.0.zip)

**Comment by MikeSchulze** [comment]
Thanks for reporting this issue.
Have you changed something in your code base before you switched to `<PackageReference Include="gdUnit4.api" Version="4.2.0-rc.202402212033" />`?

**Comment by Delsin-Yu** [comment]
No, I didn't change any source file in GdUnit4, I've only updated it according to the pop-up auto-update window.

**Comment by MikeSchulze** [comment]
Update.
It is very strange, the test run always breaks after 7 test-suite iterations.
![image](https://github.com/MikeSchulze/gdUnit4/assets/347037/e498eb6a-2669-406c-a29d-6bd90d63e6a6)


The root cause seams to be to await ProcessFrame.

```
        /// <summary>
        /// A utility to synchronize the current thread with the Godot physics thread.
        /// This can be used to await the completion of a single physics frame in Godot.
        /// </summary>
        public static SignalAwaiter SyncProcessFrame =>
            Engine.GetMainLoop().ToSignal(Engine.GetMainLoop(), SceneTree.SignalName.ProcessFrame);
```
I have no idea what goes wrong here.

**Comment by MikeSchulze** [comment]
try <PackageReference Include="gdUnit4.api" Version="4.2.0-rc.202402222030" /> it should be fixed by a crazy hack

**Comment by Delsin-Yu** [comment]
> 202402222030

Can confirm it is fixed, thanks a lot!

**Comment by MikeSchulze** [comment]
> > 202402222030
> 
> Can confirm it is fixed, thanks a lot!

Maybe you have an idea what goes wrong with the signals on my implementation?
If you have some time to look at it, I would be very grateful.

**Comment by Delsin-Yu** [comment]
I only have entry-level knowledge of the internal implementation for GodotSharp / Interop with Godot Systems, I will try to have a look later when I have time, sorry for the inconvenience.
